server/controllers/donor.controller.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const { Donor, Event, Gift, User, Profile } = require('../../config/db');
+const { messages } = require('./../helper');
+
+module.exports = {
+  create(req, res) {
+    let assignDonor = Object.assign({}, req.body, {gift_id: req.params.gift_id},
+      {user_id: req.params.id});
+    Donor.findOne({
+      where: {
+        user_id: req.params.id,
+        gift_id: req.params.gift_id
+      }
+    })
+      .then(donor => {
+        !donor &&
+        Gift.findOne({
+          where: {
+            id: req.params.gift_id,
+            is_available: true
+          }
+        }).then(gift => {
+          !gift && res.status(400).json({'message': messages.badRequest});
+          gift.dataValues.status == 'hasOneDonor' &&
```

Comment:
true == 1; // true;
true === 1; // false
its bad idea 'everywhere'. In this case i agree, but not in 
each.dataValues.user_id == every.dataValues.id == some.dataValues.id  // true;
each.dataValues.user_id === every.dataValues.id === some.dataValues.id  // false;
it causes another flow for my program..i don't agree with it at all